packages/opensrp-store/src/store/selectors.ts

```diff
@@ -11,37 +12,66 @@ interface StoreState {
     [key: string]: Dictionary;
   };
 }
-/**
- * @param {any} state - the redux store
- * @returns {Object} - the states
- */
-export function getApiToken(state: Partial<Store>): string {
-  const { extraData } = (state as StoreState)[sessionReducerName];
-  return extraData.api_token || null;
+
+interface APIFilters {
+  accessToken?: boolean;
+  apiToken?: boolean;
+  providerState?: boolean;
 }
 
-/** get Access Token from the Redux store
+/** Fetch access token from store if access_token filter val is true
  *
- * @param {any} state - the redux store
- * @returns {Object} - the states
+ * @param _ - redux store
+ * @param props - api filters object
+ * @returns {boolean}
  */
-export function getAccessToken(state: Partial<Store>): string | null {
-  const { extraData } = (state as StoreState)[sessionReducerName];
-  if (extraData.oAuth2Data && extraData.oAuth2Data.access_token) {
-    return extraData.oAuth2Data.access_token;
-  }
-  return null;
-}
 
-/** get the oAuth2 provider state parameter from the Redux store
+export const fetchAccessToken = (_: Partial<Store>, props: APIFilters): boolean | null =>
+  props.accessToken || null;
+
+/** Fetch api token from store if api_token filter val is true
  *
+ * @param _ redux store
+ * @param props api filters object
+ * @returns {boolean}
+ */
+
```

Comment:
Let's delete this space between the doc string and the function

Reply:
Prettier triggers warnings if we delete the space